mdformat_front_matters/_formatters.py

```diff
@@ -0,0 +1,184 @@
+"""Lazy-loaded formatters for different front matter formats."""
+
+from __future__ import annotations
+
+import json
+import sys
+import types
+from typing import TYPE_CHECKING, Any, TypeVar
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+T = TypeVar("T")
+
+
+class YAMLFormatter:
+    """Lazy-loaded YAML formatter using yamlfix."""
+
+    _format_fn: Callable[[str], str] | None = None
+    _yaml_module: types.ModuleType | None = None
+
+    @classmethod
+    def format(cls, content: str) -> str:
+        """Format YAML content.
+
+        Args:
+            content: Raw YAML string to format.
+
+        Returns:
+            Formatted YAML string.
+
+        """
+        if cls._format_fn is None:
+            cls._load_formatter()
+
+        try:
+            # yamlfix.fix_code returns formatted YAML
+            assert cls._format_fn is not None
+            formatted = cls._format_fn(content)
+            # yamlfix adds a YAML document start marker (---), remove it
+            if formatted.startswith("---\n"):
+                formatted = formatted[4:]
+            elif formatted.startswith("--- "):
+                # Handle scalar values like "--- Foo\n...\n"
+                formatted = formatted[4:]
+            # Remove YAML document end marker (...) if present
+            if formatted.endswith("\n...\n"):
+                formatted = formatted[:-5]
+            elif formatted.endswith("\n..."):
```

Comment:
Why not implement this in Rust?


```suggestion
            elif formatted.endswith("rusty"):
```